### PR TITLE
Go version >= 1.17

### PR DESCRIPTION
**Pull request by synarete**
Set minimal supported version of Go to 1.17. Updated github workflow accordingly.

Prefer 1.17 over 1.18 as some of our auxiliary build tools (linters) are still buggy when used with go1.18 features (namely, generics).

### CODE REVIEW

**Comment by synarete** [comment]
> Everything looks good with one minor caveat - the list of packages in go.mod looks longer that I expected (I expected new stuff in that 2nd section, but not that much). Can you please verify you ran `go mod tidy`? If so, I'll happily approve - I just want to double check the list isn't longer than necessary.

I re-run `go mod tidy` and got same result, both with `go1.17` and `go1.18`. May need to do deeper dive into https://go.dev/ref/mod#go-mod-file-require but I would prefer to do so in a separate PR.

**Review by phlogistonjohn — APPROVED** [review]
LGTM

**Comment by phlogistonjohn** [comment]
I'll wait approx. one day before merging for other feedback. I will ignore the centos ci status for this PR and merge w/o waiting for success from that CI.

File: tests/integration/path_permissions.go
Comment (anoopcs9) [inline tests/integration/path_permissions.go:1]: #243 includes this change.

File: tests/integration/resource_update_test.go
Comment (anoopcs9) [inline tests/integration/resource_update_test.go:3]: #243 includes this change.

File: internal/planner/configuration.go
Comment (anoopcs9) [inline internal/planner/configuration.go:88]: This is same as #247 , right?

**Review by anoopcs9 — CHANGES_REQUESTED** [review]
Few changes from this PR already exists as separate PRs(see below comments). I prefer to consider those older PRs and accordingly modify the current one.

**Review by phlogistonjohn — APPROVED** [review]
LGTM, thanks.

**Review by anoopcs9 — APPROVED** [review]
lgtm.

**Pull request merged**
